### ISSUE TITLE
LogReceiverWebServiceTarget should remove its use of LogEventInfoBuffer

### ISSUE THREAD

**Issue by snakefoot** [NLog/NLog#4081]
Targets should not implement their own batching. Instead they should do one of the following:

- Recommend wrapping using AsyncWrapper or BufferingWrapper and override `Write(IList<AsyncLogEventInfo>)`
- Inherit from AsyncTaskTarget (for async / await)

Alternative move LogEventInfoBuffer into the WCF-nuget-package for it to die there.

**Comment by noamyogev84** [comment]
Hi.
I'd like to take this.
Any hint points?

**Comment by snakefoot** [comment]
I think the easy solution is just to copy the `LogEventInfoBuffer`-code, and make it an internal-class in the `NLog.Wcf`-project (Along with removing obsolete-attribute). Then change these lines:

```c#
        #pragma warning disable CS0618 // Type or member is obsolete
        private readonly LogEventInfoBuffer buffer = new LogEventInfoBuffer(10000, false, 10000);
        #pragma warning restore CS0618 // Type or member is obsolete
```

Into just this (Removing suppression of compiler warnings):

```c#
        private readonly LogEventInfoBuffer buffer = new LogEventInfoBuffer(10000, false, 10000);`
```

**Comment by snakefoot** [comment]
Resolved with #4126
